Slim/Player/StreamingController.pm

```diff
@@ -2230,6 +2230,13 @@ sub playerStopped {
 		}
 	}
 
+	# If queue finished playing then reset song queue
```

Comment:
I'm sorry for the lack of feedback. Time's running... So I spent some time playing with this. Don't you think your code should go inside the `else` clause, as we only should be at the end when there's only one single track left (the current) in the `songqueue`?

Would you have a better idea than I do as to when `playerStopped()` would be called? It's my understanding that this would eg. happen when the player runs out of buffer data. But this could have other reasons than end of stream, I believe.

Reply:
I'm not even sure you'd need the "is this the last song" check, or whether you could simply `resetSongqueue()` if `@{$queue} <= 1`....

Reply:
The check might have been needed after all. See https://forums.lyrion.org/forum/user-forums/logitech-media-server/1780688.